assets/js/tutorials/trade-in-the-dex.js

```diff
@@ -0,0 +1,278 @@
+  // compensate.
+
+  const offers = orderbook_resp.result.offers
+  const want_amt = BigNumber(we_want.value)
+  let running_total = BigNumber(0)
+  if (!offers) {
+    block.find(".output-area").append(`<p>No Offers in the matching book.
+                 Offer probably won't execute immediately.</p>`)
+  } else {
+    for (const o of offers) {
+      if (o.quality <= proposed_quality) {
+        block.find(".output-area").append(
+            `<p>Matching Offer found, funded with ${o.owner_funds}</p>`)
+        running_total = running_total.plus(BigNumber(o.owner_funds))
+        if (running_total >= want_amt) {
+          block.find(".output-area").append("<p>Full Offer will probably fill.</p>")
+          break
+        }
+      } else {
+        // Offers are in ascending quality order, so no others after this
+        // will match, either
+        block.find(".output-area").append(`<p>Remaining orders too expensive.</p>`)
+        break
+      }
+    }
+    block.find(".output-area").append(`<p>Total matched: ${Math.min(running_total, want_amt)}</p>`)
+    if (running_total > 0 && running_total < want_amt) {
+      block.find(".output-area").append(`<p>Remaining ${want_amt - running_total} would probably be
+                   placed on top of the order book.</p>`)
+    }
+  }
+
+  if (running_total == 0) {
+    // If part of the Offer was expected to cross, then the rest would be placed
+    // at the top of the order book. If none did, then there might be other
+    // Offers going the same direction as ours already on the books with an
+    // equal or better rate. This code counts how much liquidity is likely to be
+    // above ours.
+
+    // Unlike above, this time we check for Offers going the same direction as
+    // ours, so TakerGets and TakerPays are reversed from the previous
+    // book_offers request.
+    let orderbook2_resp
+    try {
+      orderbook2_resp = await api.request({
+        "command": "book_offers",
+        "taker": address,
+        "ledger_index": "current",
+        "taker_gets": we_spend,
+        "taker_pays": we_want
+      })
+      block.find(".output-area").append(
+        `<pre><code>${pretty_print(orderbook2_resp.result)}`)
+    } catch(err) {
+      show_error(block, err)
+      block.find(".loader").hide()
+      return
+    }
+
+    // Since TakerGets/TakerPays are reversed, the quality is the inverse.
+    // You could also calculate this as 1/proposed_quality.
+    const offered_quality = BigNumber(we_want.value) / BigNumber(we_spend.value)
+
+    const offers2 = orderbook2_resp.result.offers
+    let running_total2 = BigNumber(0)
+    if (!offers2) {
+      block.find(".output-area").append(`<p>No similar Offers in the book. Ours would be the first.</p>`)
+    } else {
+      for (const o of offers2) {
+        if (o.quality <= offered_quality) {
+          block.find(".output-area").append(`<p>Existing offer found, funded with ${o.owner_funds}</p>`)
```

Comment:
Should this be "asking for __amount of the currency they're asking to exchange__" instead of "funded with"? (Because the currency we care about is the "we_want" currency)

Reply:
No, in this case, we are looking at how much the other traders are able to sell (funded with) of the same thing we're offering to sell, at a better rate than we are offering.